### ISSUE TITLE
"Rate Limit reached" on valid OpenAI account

### ISSUE THREAD

**Issue by romanr** [Blarc/ai-commits-intellij-plugin#93]
Plugin thinks about 30 sec and then outputs:

```
Error occurred: Rate limit reached for default-gpt-3.5-turbo in organization org-qhdxxxxslxlioFX7inZ on tokens per min. Limit: 90000 / min. Current: 0 / min. Contact us through our help center at help.openai.com if you continue to have issues.
```
Tryied different prompts - same result.
OpenAI account  uses pay as you go billing, and works fine with any other intgration.

**Comment by krrishdholakia** [comment]
Hey @romanr if you're able create multiple azure embedding instances - you could use LiteLLM's openai-compatible server  to loadbalance across from them: 

**Step 1: Put your instances in a `config.yaml`**
```yaml
model_list:
  - model_name: zephyr-beta
    litellm_params:
        model: huggingface/HuggingFaceH4/zephyr-7b-beta
        api_base: http://0.0.0.0:8001
  - model_name: zephyr-beta
    litellm_params:
        model: huggingface/HuggingFaceH4/zephyr-7b-beta
        api_base: http://0.0.0.0:8002
  - model_name: zephyr-beta
    litellm_params:
        model: huggingface/HuggingFaceH4/zephyr-7b-beta
        api_base: http://0.0.0.0:8003
```

**Step 2: Install LiteLLM**

```bash
$ pip install litellm
```

**Step 3: Start litellm proxy w/ config.yaml**

```bash
$ litellm --config /path/to/config.yaml
```

Docs: https://docs.litellm.ai/docs/simple_proxy

**Comment by romanr** [comment]
I just wanted a plugin that suggests commit messages, do I have to sign up for Azure and spin up VMs for that?

**Comment by Blarc** [comment]
Is this still an issue. Would you mind trying please? The plugin uses [openai-kotlin](https://github.com/aallam/openai-kotlin) for making requests to OpenAI API. The dependency was updated in the latest plugin version.

**Comment by Blarc** [comment]
Can you make requests to OpenAI API with your API key: https://platform.openai.com/docs/api-reference/making-requests ? There are different OpenAI plans and some plans do not give credits for calling the API directly.

I think you can't call the API with ChatGPT Plus API key.

**Comment by whjvenyl** [comment]
> Is this still an issue? Would you mind trying again please? The plugin uses [openai-kotlin](https://github.com/aallam/openai-kotlin) for making requests to OpenAI API. The dependency was updated in the latest plugin version.

yes it is

- new plugin installation out of the intellij marketplace
- replaced the token with a newly created on the openai platform
- error message as shown above

**Comment by Blarc** [comment]
> > Is this still an issue? Would you mind trying again please? The plugin uses [openai-kotlin](https://github.com/aallam/openai-kotlin) for making requests to OpenAI API. The dependency was updated in the latest plugin version.
> 
> yes it is
> 
> * new plugin installation out of the intellij marketplace
> * replaced the token with a newly created on the openai platform
> * error message as shown above



> Can you make requests to OpenAI API with your API key: https://platform.openai.com/docs/api-reference/making-requests ? There are different OpenAI plans and some plans do not give credits for calling the API directly.
> 
> I think you can't call the API with ChatGPT Plus API key.
> 
> edit: Check also https://community.openai.com/t/can-i-have-the-same-service-using-a-chatgpt-plus-or-an-openai-subscription/370603

Can you please try the things I have written in the previous comment.

**Comment by dsum27** [comment]
I recently tried out this plugin and encountered a rate limit error initially, which seemed to be linked to the default OpenAI settings. It turns out the issue was related to my account being a free one without any billing details or a minimum balance. Once I updated my account with billing information and added a minimum of $5, the plugin started functioning smoothly without any rate limit errors.

Great tool. Thanks!
